src/site/content/en/blog/building-a-button-component/index.md

```diff
@@ -0,0 +1,1274 @@
+  <input type="button" value="<input>"/>
+  <button>Icon <span data-icon="cloud"></span></button>
+  <button type="submit">Submit</button>
+  <button type="button">Type Button</button>
+  <button type="reset">Reset</button>
+  <button disabled>Disabled</button>
+  <button class="btn-custom btn-large" type="button">Large Custom</button>
+  <input type="file">
+</form>
+```
+
+The matrix of combinations is pretty overwhelming at this point. Between button
+types, pseudo classes, and being in or out of a form, there are  over 20
+combinations of buttons. It’s a good thing CSS can help us articulate each of
+them clearly!
+
+## Accessibility
+
+Button elements are naturally accessible but there are a few common
+enhancements.
+
+### Hover and focus together
+
+I like to group `:hover` and `:focus` together with the `:is()` functional
+pseudo selector. This helps ensure my interfaces are always considering keyboard
+and assistive technology styles.
+
+```css
+button:is(:hover, :focus) {
+  …
+}
+```
+
+<figure>
+  {% Video 
+    src="video/vS06HQ1YTsbMKSFTIPl2iogUQP73/FwNnm4ExQWz2FO1GVa7j.mp4",
+    autoplay="true",
+    loop="true",
+    muted="true"
+  %}
+  <figcaption>
+    Try a [demo](https://codepen.io/argyleink/pen/powMQgx)
+  </figcaption>
+</figure>
+
+### Interactive focus ring
+
+I like to animate the focus ring for keyboard and assistive technology users. I
+accomplish this by animating the outline away from the button by 5px, but only
+when the button is not active. This creates an effect that makes the focus ring
+shrink back to the button size when pressed.
+
+```css
+:where(button, input):where(:not(:active)):focus-visible {
+  outline-offset: 5px;
+}
+```
+
+{% Video 
+  src="video/vS06HQ1YTsbMKSFTIPl2iogUQP73/iMI3C1a3ktt1f3vZGscB.mp4",
+  autoplay="true",
+  loop="true",
+  muted="true"
+%}
+
+### Ensuring passing color contrast
+
+There are at least four different color combinations across light and dark that
+need consideration of color contrast: button, submit button, reset button, and
+disabled button. [VisBug](https://a.nerdy.dev/gimme-visbug) is used here to
+inspect and show all their scores at once:
+
+{% Video 
+  src="video/vS06HQ1YTsbMKSFTIPl2iogUQP73/Wg0EzKQgCHMJin5ujYus.mp4",
+  autoplay="true",
+  loop="true",
+  muted="true"
+%}
+
+### Hiding icons from folks who can't see
+
+When creating an icon button, the icon should provide visual support to the
+button text. This also means the icon is not valuable to someone with sight
+loss. Fortunately the browser provides a way to hide items from screen-reader
+technology so people with sight loss aren't bothered with decorative button
+images:
+
+```html
+<button>
+  <svg … aria-hidden="true">...</svg>
+  Icon Button
+</button>
+```
+
+<figure>
+  {% Img 
+    src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/WCww9HPVdcaoJNLQB8Sq.png", 
+    alt="Chrome DevTools showing the accessibility tree for the button. The tree ignores the button image because it has aria-hidden set to true",
```

Comment:
```suggestion
    alt="Chrome DevTools showing the accessibility tree for the button. The tree ignores the button image because it has aria-hidden set to true.",
```